### PR TITLE
CB-2791 Redbeams cf template NPE because of cidr null

### PR DESCRIPTION
**Pull request by gergopapi2**

### CODE REVIEW

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java [hunk 1]
```diff
@@ -21,10 +21,9 @@
         return parameters;
     }
 
-    public Map<String, Object> addVpcParameters(Map<String, Object> parameters, EnvironmentNetworkResponse environmentNetworkResponse) {
-        parameters.put(VPC_ID, String.join(",", environmentNetworkResponse.getAws().getVpcId()));
-        parameters.put(VPC_CIDR, environmentNetworkResponse.getNetworkCidr());
+    public Map<String, Object> addVpcParameters(Map<String, Object> parameters, DetailedEnvironmentResponse environmentResponse) {
+        parameters.put(VPC_CIDR, environmentResponse.getSecurityAccess().getCidr());
+        parameters.put(VPC_ID, environmentResponse.getNetwork().getAws().getVpcId());
```
Comment (jadair-cloudera) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java]: I am surprised to see getAws() in a class and package that does not seem to be AWS-specific.

Comment (gergopapi2) [reply redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java]: Yep, you are right. So I cured that place.

File: redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java [hunk 2]
```diff
@@ -58,6 +57,8 @@
 
     private static final DBStackStatus NEW_STATUS = new DBStackStatus();
 
+    private static final List<CloudPlatform> SUPPORTED_CLOUD_PLATFORMS = List.of(CloudPlatform.AWS);
```
Comment (meowfaceman) [inline redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java]: Can you add Azure to this list as well? I'll be checking in Azure support shortly.

File: redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java [hunk 3]
```diff
@@ -118,24 +119,43 @@ private void setRegion(DBStack dbStack, DetailedEnvironmentResponse environment)
...
-            source.getDatabaseServer().setCloudPlatform(CloudPlatform.valueOf(cloudPlatform));
+        dbStack.setCloudPlatform(cloudPlatformRequest);
+        dbStack.setPlatformVariant(cloudPlatformRequest);
+
+        return cloudPlatform;
+    }
+
+    private void checkCloudPlatformIsSupported(CloudPlatform cloudPlatform) {
+        if (!SUPPORTED_CLOUD_PLATFORMS.contains(cloudPlatform)) {
+            throw new BadRequestException(String.format("Cloudplatform %s not supported yet.", cloudPlatform));
```
Comment (meowfaceman) [inline redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java]: nit: "Cloud platform"

File: redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java [hunk 4]
```diff
@@ -118,24 +119,43 @@ private void setRegion(DBStack dbStack, DetailedEnvironmentResponse environment)
...
+        }
+    }
+
+    private void checkCloudPlatformsMatch(String cloudPlatformEnvironment, String cloudPlatformRequest) {
+        if (!cloudPlatformEnvironment.equals(cloudPlatformRequest)) {
+            throw new BadRequestException(String.format(
+                    "Cloudplatform of the request %s and the environment %s do not match.", cloudPlatformRequest, cloudPlatformEnvironment));
```
Comment (meowfaceman) [inline redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java]: nit: "Cloud platform"

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java [hunk 5]
```diff
@@ -21,10 +23,18 @@
...
+        parameters.put(VPC_ID, getVpcId(environmentResponse, cloudPlatform));
         return parameters;
     }
 
+    private String getVpcId(DetailedEnvironmentResponse environmentResponse, CloudPlatform cloudPlatform) {
+        switch (cloudPlatform) {
+            case AWS:
+                return environmentResponse.getNetwork().getAws().getVpcId();
+            default:
+                throw new RedbeamsException(String.format("Support for cloudplatform %s not yet added", cloudPlatform.name()));
```
Comment (jadair-cloudera) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java]: NIT: "cloud platform"

File: redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java (hunk 2 above)
Comment (gergopapi2) [reply redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java]: Yepp, I added.

File: redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java (hunk 4 above)
Comment (gergopapi2) [reply redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java]: Nice catch, thanks!

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java (hunk 5 above)
Comment (gergopapi2) [reply redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java]: Thanks for catching this!

**Pull request merged**